File: agora_plugin/android/src/main/java/com/agora/agora_plugin/AgoraRenderViewFactory.java
package com.agora.agora_plugin;

import android.content.Context;
import android.view.SurfaceView;

import java.util.HashMap;

import io.agora.rtc.RtcEngine;
import io.flutter.plugin.common.MessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

public class AgoraRenderViewFactory extends PlatformViewFactory {
    private HashMap<String, SurfaceView> mRendererViews = new HashMap<>();

    public void addView(SurfaceView view, int id) {
        mRendererViews.put("" + id, view);
    }

    public void removeView(int id) {
        mRendererViews.remove("" + id);
    }

    public SurfaceView getView(int id) {
        return mRendererViews.get("" + id);
    }

    public AgoraRenderViewFactory(MessageCodec<Object> createArgsCodec) {
        super(createArgsCodec);
    }

    @Override
    public PlatformView create(Context context, int id, Object o) {
        SurfaceView view = RtcEngine.CreateRendererView(context.getApplicationContext());
        AgoraRendererView rendererView = new AgoraRendererView(view, id);
        addView(view, id);
        return rendererView;
    }
}

File: agora_plugin/android/src/main/kotlin/com/example/agora_plugin/AgoraRenderViewFactory.kt
package com.example.agora_plugin

import android.content.Context
import android.view.SurfaceView
import io.agora.rtc.RtcEngine
import io.flutter.plugin.common.MessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import java.util.*

class AgoraRenderViewFactory(createArgsCodec: MessageCodec<Any?>?) : PlatformViewFactory(createArgsCodec) {
    private val mRendererViews = HashMap<String, SurfaceView>()
    fun addView(view: SurfaceView, id: Int) {
        mRendererViews["" + id] = view
    }

    fun removeView(id: Int) {
        mRendererViews.remove("" + id)
    }

    fun getView(id: Int): SurfaceView? {
        return mRendererViews["" + id]
    }

    override fun create(context: Context, id: Int, o: Any): PlatformView {
        val view = RtcEngine.CreateRendererView(context.applicationContext)
        val rendererView = AgoraRendererView(view, id)
        addView(view, id)
        return rendererView
    }
}